pkg/component/controller/workerconfig/reconciler_test.go

```diff
@@ -52,7 +52,7 @@ type kubeletConfig = kubeletv1beta1.KubeletConfiguration
 
 // TODO: simplify it somehow, it is hard to read and to modify, both tests and implementation
 func TestReconciler_Lifecycle(t *testing.T) {
-	createdReconciler := func(t *testing.T) (*Reconciler, testutil.FakeClientFactory) {
+	createdReconciler := func(t *testing.T) (*Reconciler, *testutil.FakeClientFactory) {
```

Comment:
The same goes for these.

Maybe they don't need to be paired like this? Perhaps there could be something like:

```go
func createReconciler(t *testing.T, clientFactory ClientGetter) *Reconciler {
	k0sVars, err := config.NewCfgVars(nil, t.TempDir())
	require.NoError(t, err)
	reconciler, err := NewReconciler(
		k0sVars,
		&v1beta1.ClusterSpec{
			API: &v1beta1.APISpec{},
			Network: &v1beta1.Network{
				ClusterDomain: "test.local",
				ServiceCIDR:   "99.99.99.0/24",
			},
		},
		clients,
		&leaderelector.Dummy{Leader: true},
		true,
	)
	require.NoError(t, err)
	reconciler.log = newTestLogger(t)
	return reconciler
}
```

and:

```go
		clients := createFakeClientFactory()
		reconciler := createReconciler(t, clients)
```

Reply:
Anyways, done in 2c790e52779951665636facdb31b39231a10cc5c. :sweat_smile: